src/ApiV2.js

```diff
@@ -179,7 +179,7 @@ export default class ApiV2Client {
       const maybeErrorData = idx(e, _ => _.response.data.errors.length);
       if (maybeErrorData) {
         result = e.response.data;
-      } else if (e.code === 'ECONNREFUSED' || e.code === 'ENOTFOUND' || e.code === 'ETIMEDOUT') {
+      } else if (e.code.match(/E[A-Z]+/)) {
```

Comment:
This fails for errors where `e.code` is not a string.

I think we probably should eliminate the `There was a problem understanding the server.` blanket errors altogether and just rethrow any errors that don't result in a `ApiV2Error` (handled below). Then we won't have to specifically whitelist network errors.

Reply:
Yep would be better I guess. So is this PR still relevant?

Reply:
I would welcome a PR with that change (or just with changes to this PR), if you'd like to make it. No worries, if not, I can also tackle it when I have some time.